pkg/check/zones.go

```diff
@@ -131,7 +134,19 @@ func CheckZoneTags(ctx *CheckContext) error {
 
 	// Perform check if FailureDomains defined.  We need 2 or more to require tags.
 	klog.V(4).Info("Checking failure domains.")
-	if fds := inf.Spec.PlatformSpec.VSphere.FailureDomains; len(fds) > 1 {
+
+	var fds []v1.VSpherePlatformFailureDomainSpec
+
+	// In existing UPI clusters the VSphere field in the infra spec
+	// may not be defined.
+	if inf.Spec.PlatformSpec.VSphere != nil {
```

Comment:
It seems that `inf.Spec.PlatformSpec.VSphere` should be `inf.Spec.PlatformSpec.VSphere.FailureDomains`, when in inf vsphere
```console
platformSpec:
  type: VSphere
  vsphere: {}
``` 
actually `inf.Spec.PlatformSpec.VSphere` is `&{[] [] {{[]  []} {[]  []}}}` not `nil`

Reply:
@gnufied Yeah, `inf.Spec.PlatformSpec.VSphere` is never nil but it is `inf.Spec.PlatformSpec.VSphere.FailureDomains` which could be nil, we are fine, just a bit log output issue, so when `vsphere: {}` the info `VSphere Infrastructure spec is empty, no FailureDomains configured. Skipping check.`  doesn't print however seems we want to print it when `vsphere: {}`.